app/controllers/admin/email_controller.rb

```diff
@@ -69,6 +69,19 @@ def send_digest
     end
   end
 
+  def smtp_should_reject
+    params.require(:from)
+    params.require(:to)
+    # These strings aren't localized; they are sent to an anonymous SMTP user.
+    if User.find_by_email(params[:from]).nil? && !SiteSetting.enable_staged_users
```

Comment:
I usually prefer `User.exists?(email: Email.downcase(params[:from]))` to test if an user exists with that email.